Sources/Widgets/Widgets3D/TransformControlsWidget/index.js

```diff
@@ -0,0 +1,181 @@
+  mat3.fromQuat(m3, transformState.getRotation());
+
+  [sx, hx].forEach((state) => {
+    state.setDirection(m3.slice(0, 3));
+    state.setUp(m3.slice(3, 6).map((c) => -c));
+    state.setRight(m3.slice(6, 9));
+  });
+
+  [sy, hy].forEach((state) => {
+    state.setDirection(m3.slice(3, 6));
+    state.setUp(m3.slice(6, 9));
+    state.setRight(m3.slice(0, 3));
+  });
+
+  [sz, hz].forEach((state) => {
+    state.setDirection(m3.slice(6, 9));
+    state.setUp(m3.slice(3, 6));
+    state.setRight(m3.slice(0, 3));
+  });
+}
+
+// ----------------------------------------------------------------------------
+// Factory
+// ----------------------------------------------------------------------------
+
+function vtkTransformControlsWidget(publicAPI, model) {
+  model.classHierarchy.push('vtkTransformControlsWidget');
+
+  // --- Widget Requirement ---------------------------------------------------
+
+  model.behavior = widgetBehavior;
+  model.widgetState = stateGenerator();
+
+  model.methodsToLink = [
+    'scaleInPixels',
+    'activeScaleFactor',
+    'useActiveColor',
+    'activeColor',
+  ];
+
+  publicAPI.getRepresentationsForViewType = (viewType) => {
+    switch (viewType) {
+      default:
+        return [
+          {
+            builder: vtkTranslateTransformHandleRepresentation,
+            labels: ['translateHandles'],
+            initialValues: {
+              radius: TRANSLATE_HANDLE_RADIUS,
+              glyphResolution: 12,
+              coneSource: {
+                radius: 8,
+                height: 0.05,
+                direction: [0, 1, 0],
+              },
+            },
+          },
+          {
+            builder: vtkScaleTransformHandleRepresentation,
+            labels: ['scaleHandles'],
+            initialValues: {
+              radius: SCALE_HANDLE_RADIUS,
+              glyphResolution: 12,
+              cubeSource: {
+                xLength: SCALE_HANDLE_CUBE_SIDE_LENGTH,
+                yLength:
+                  SCALE_HANDLE_CUBE_SIDE_LENGTH / SCALE_HANDLE_PIXEL_SCALE,
+                zLength: SCALE_HANDLE_CUBE_SIDE_LENGTH,
+              },
+            },
+          },
+          {
+            builder: vtkRotateTransformHandleRepresentation,
+            labels: ['rotateHandles'],
+          },
+        ];
+    }
+  };
+
+  publicAPI.updateHandleVisibility = () => {
+    model.widgetState
+      .getStatesWithLabel('translateHandles')
+      .forEach((state) => {
+        state.setVisible(model.mode === 'translate');
+      });
+    model.widgetState.getStatesWithLabel('scaleHandles').forEach((state) => {
+      state.setVisible(model.mode === 'scale');
+    });
+    model.widgetState.getStatesWithLabel('rotateHandles').forEach((state) => {
+      state.setVisible(model.mode === 'rotate');
+    });
+  };
+
+  const parentSetMode = publicAPI.setMode;
+  publicAPI.setMode = (mode) => {
+    if (parentSetMode(mode)) {
+      publicAPI.updateHandleVisibility();
+    }
+  };
```

Comment:
as a reminder, you alternatively also have the `model._onModeChanged()` convenient method you can use.